lib/modules/datasource/dotnet-sdk/index.ts

```diff
@@ -0,0 +1,83 @@
+import is from '@sindresorhus/is';
+import { cache } from '../../../util/cache/package/decorator';
+import type { HttpResponse } from '../../../util/http/types';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { DotnetSdkReleases, DotnetSdkReleasesIndex } from './types';
+
+export class DotnetSdkDatasource extends Datasource {
+  static readonly id = 'dotnet-sdk';
+
+  constructor() {
+    super(DotnetSdkDatasource.id);
+  }
+
+  override readonly customRegistrySupport = false;
+
+  override readonly defaultRegistryUrls = [
+    'https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json',
+  ];
+
+  @cache({
+    namespace: `datasource-${DotnetSdkDatasource.id}`,
+    key: DotnetSdkDatasource.id,
+  })
+  async getReleases(_: GetReleasesConfig): Promise<ReleaseResult | null> {
+    let result: ReleaseResult | null = null;
+
+    let raw: HttpResponse<DotnetSdkReleasesIndex> | null = null;
+    try {
+      raw = await this.http.getJson<DotnetSdkReleasesIndex>(
+        this.defaultRegistryUrls[0]
+      );
+    } catch (err) {
+      this.handleGenericErrors(err);
+    }
+
+    const body = raw?.body;
+    if (body) {
+      const { 'releases-index': releasesIndex } = body;
+      const releases = (
+        await Promise.all(
+          releasesIndex
```

Comment:
Make sure this is an array, e.g. using `coerceArray()`

Reply:
we should use the new JSON Schema validation feature here

we should also cache the sub release files much longer and only get h them when they changed, can be easily done by checking the main release file.

the old EOL versions will no longer change

Reply:
I didn't realise that had landed! Happy to move over to it.

Reply:
I've moved over to JSON schema validation (runtime type checking), added a cache for the release channel fetching, and set the cache time to 1440 minutes (1 day) for both caches.